src/TreeSupport.cpp

```diff
@@ -562,26 +562,36 @@ void TreeSupport::generateContactPoints(const SliceMeshStorage& mesh, std::vecto
 
     //First generate grid points to cover the entire area of the print.
     AABB bounding_box = mesh.bounding_box.flatten();
-    //We want to create the grid pattern at an angle, so compute the bounding box required to cover that angle.
-    constexpr double rotate_angle = 22.0 / 180.0 * M_PI; //Rotation of 22 degrees provides better support of diagonal lines.
+    // We want to create the grid pattern at an angle, so compute the bounding
+    // box required to cover that angle.
+    // Rotation of 22 degrees provides better support of diagonal lines.
+    constexpr double rotate_angle = 22.0 / 180.0 * M_PI;
     const Point bounding_box_size = bounding_box.max - bounding_box.min;
-    AABB rotated_bounding_box; //Bounding box is rotated around the lower left corner of the original bounding box, so translate everything to 0,0 and rotate.
-    rotated_bounding_box.include(Point(0, 0));
-    rotated_bounding_box.include(rotate(bounding_box_size, -rotate_angle));
-    rotated_bounding_box.include(rotate(Point(0, bounding_box_size.Y), -rotate_angle));
-    rotated_bounding_box.include(rotate(Point(bounding_box_size.X, 0), -rotate_angle));
-    AABB unrotated_bounding_box; //Take the AABB of that and rotate back around the lower left corner of the original bounding box (still 0,0 coordinate).
-    unrotated_bounding_box.include(rotate(rotated_bounding_box.min, rotate_angle));
-    unrotated_bounding_box.include(rotate(rotated_bounding_box.max, rotate_angle));
-    unrotated_bounding_box.include(rotate(Point(rotated_bounding_box.min.X, rotated_bounding_box.max.Y), rotate_angle));
-    unrotated_bounding_box.include(rotate(Point(rotated_bounding_box.max.X, rotated_bounding_box.min.Y), rotate_angle));
+
+    // Store centre of AABB so we can relocate the generated points
+    const auto centre = bounding_box.getMiddle();
+    const auto sin_angle = std::sin(rotate_angle);
+    const auto cos_angle = std::cos(rotate_angle);
+    // Calculate the dimensions of the AABB of the mesh AABB after being rotated
+    // by `rotate_angle`. Halve the dimensions since we'll be using it as a +-
+    // offset from the centre of `bounding_box`.
+    const auto rotated_dims = Point(
+        bounding_box_size.X * cos_angle + bounding_box_size.Y * sin_angle,
+        bounding_box_size.X * sin_angle + bounding_box_size.Y * cos_angle) / 2;
```

Comment:
This can't be correct. You're only rotating one point, so somewhere it's going to map to 0. For instance, if I put an angle of 110 in there and a bounding box of 100x300, it'll output a rotated bounding box of 247x-9.

Reply:
This will only work for angles <90 as the either the sine or cosine will be <0. This isnt a problem for this code since the angle is hardcoded as 22 degrees but will obviously be an issue if that angle becomes a user set angle. I'll add a comment to make it more clear.

What i'm trying to achieve is better shown below. Left is the original bounding box, middle is rotated, right is the AABB of the rotated bounding box. The `rotated_dims` calculation directly gives us the dimensions of the right hand box (halved since i'm using it as an offset from the centre of the AABB)

![image](https://user-images.githubusercontent.com/4530480/49705108-ac208700-fc56-11e8-800a-b1b4130e4c08.png)